### ISSUE TITLE
Linux wallet does not give access to all keys

### ISSUE THREAD

**Issue by InMyImage** [valiant1x/intensecoinwallet#16]
Using  Intensecoin-linux-x86_64.AppImage displays what I assume is the public key to use in minining software like xmr-stak, and will allow you complete transactions, but it does not show the spend key or private key.

Based on info for the non-linux wallets, there are files that the keys are in, but the AppImage does not create any files in the subdirectory where I put the AppImage, so... where are the keys stored and how do I get them?

I started mining, but am not comfortable continuing until I have all the keys.

**Comment by valiant1x** [comment]
Can you please go into more detail on how the keys are not being displayed properly? How are you attempting to view said keys?

Keys are stored in `~/.intensecoin/.*wallet` by default

**Comment by InMyImage** [comment]
The .wallet file is not hman readable so it doesn’t make the keys available for use except via the IntenseCoin wallet application.

 I’m currently mining using xmr-stak at hashvault.pro and hoping that the key displayed at the top of the wallet screen under “Your wallet” is indeed the Public Key that other people and applications like mining pools would use to send me Intense Coins because the box label does not say which key it is.

All of my other coin wallets have 3 keys:
Public Key for others to send me coins.
Private Spend for me to be able to spend my coins
Private View which since I’m new I’m honestly not sure what to do with this yet since I’m just accumulating by mining right now.

I can successfully save “keys” from the file menu where the options are:
Backup wallet
Save wallet keys
Export key
Export tracking key 

Save wallet keys and backup wallet both generate the encrypted .wallet file which is not readable (not that I have not encrypted my wallet so the lock under they for “Your wallet” is currently unlocked and says “Wallet not encrypted” beside it.

Export key and Export tracking key both open a dialog displaying an all caps alpha numeric string, whereas “Your wallet” is mixed case alphanumeric like all of the keys for my non-Intense keys. Neither of these dialogs indicate exactly which key it is “exporting” although a logical leap can be made indicating that “Export tracking key” may be the private view key. As an aside note that both of these dialogs labels say Import.

Both of the exported “tracking key” also has a string snippet of around 30 zeros in the middle of it.

Give the all upper case exported values, neither would be considered globally unique values.

The Send Intense screen does not show the key being used to send coins and also does not have a field for setting the desired key if it is different than the one currently loaded.

Let me know if you need more info

Bill

**Comment by InMyImage** [comment]
I hate bumping things, but I’m surprised that after 5 days I still haven’t received an answer to whether the key in the “your wallet” field is the public key or the private spend key. I’d expect that one to be simple to answer even if there needs to be a “fix”to make the other keys human readable.

I’d like to continue mining this coin, but without getting confirmation on which key is being dispaly, I can’t continue throwing hash time at it.

I’m not as concerned about being able to extract the spend oe view keys right now because I don’t have any immediate plans to spend them, but the public key is critical.

**Comment by cloudrck** [comment]
I wish there was some sort of GUI standard for CryptoNote wallets. But anyway, you have to read the [spec ](https://cryptonote.org/cns/cns007.txt)and [whitepaper](https://cryptonote.org/whitepaper.pdf) for CryptoNote.

All of the keys are stored in the .wallet. The other menu options extracts the keys from the wallet.

Since IntenseCoin currently uses CryptoNote wallet and is untraceable, it can be a bit confusing.

Notable excerpts from the spec

>Each user has two pairs of (private, public) permanent keys by
   default. The public parts of the keys are represented as user
   address. In order to send money, one needs two public keys corresponding to
   the two secret keys the receiver possesses.

This is broken down as follows.

> Spend key is a pair of secret keys corresponding to the public keys
   in a CryptoNote address. Spend key is required to sign transactions.

So all of your private keys are "spend keys" (Secret key a & Secret key b)

>   View key consists of one secret key corresponding to a public key in
   a CryptoNote address and one public key from the same address. View
   key allows its holder to identify incoming transactions,
   circumventing unlinkability.

More [info here](https://monero.stackexchange.com/questions/954/how-to-use-the-view-key-to-see-amount-sent-to-cold-storage)

> CryptoNote addresses are serialized to strings using CryptoNote
   Base58 encoding scheme. The data encoded consists of three parts:

> public address prefix (used to distinguish the addresses of 
        different currencies),

>a pair of public keys,

So One wallet address consists of two public keys which were derived from your two secret keys.

A tracking key

>tracking key is a pair (a, B) of private and public ec-key (where B = bG and a 6= b);

Recipient gives part of his *private* key to the network. This is called "tracking key".
The network can verify the transaction but still can't find the actual wallet address of the receiver. Thus the anonymity part.

Exporting the "Wallet keys" creates a machine friendly representation of your spend keys.

>I’d expect that one to be simple to answer even if there needs to be a “fix”to make the other keys human readable.

You should research the basics of encryption to understand why all you actually need is the wallet key pairs. Everything is is derived from these keys. CryptoNote, and in turn [IntenseCoin uses Keccak](https://cryptonote.org/cns/cns008.txt) to generate various keys it utilizes.

If you want a "human friendly" representation of your keys, you can generate it. All you need are your private & public key pairs. I plan on writing a Python script to do just that. Something similar [to this.](https://xmr.llcoins.net/)

[One specific example. ](https://github.com/moneroexamples/spendkey)

You can generate a [mnemonic seed key](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#Abstract)

TL:DR. Everything is exported in your .wallet file. When you export your keys, it's in a machine-friendly format. You can use command line tools to extract it to a human readable format. View keys, spend keys, tracking keys are nothing more than variations and combinations/hashed from your private keys.

**Comment by InMyImage** [comment]
Thanks, in some way that both helps clear things up and muddies them at the same time. Reminds me that develophave very little understanding of usability and why wide adoption of crypto will continue to take time, especially given the wild gyrations of the market right now.
